the-collab-lab/tcl-29-smart-shopping-list
# Zi nr sort items

## Description

We sorted items by nextPurchase estimate in ascending order (soon -> kind of soon -> not so soon -> inactive). If multiple items have the same day of nextPurchase, we sort them in alphabetical order. We gave distinct colors to differentiate between the nextPurchase ranges and set the aria-label accordingly.

## Related Issue

Closes #12 

## Acceptance Criteria


## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

1. Pull down `zi-nr-sort-items` branch and run `npm start`
2. Create items of each soon, kind of soon and not soon. Navigate to list-view, make sure all of the newly created items are inactive (colored grey).
3. On firebase, change nextPurchase and make sure the items are sorted in ascending order (soon -> kind of soon -> not so soon -> inactive).
4. On firebase, change nextPurchase on multiple items to be the same number. Make sure they are sorted alphabetically.
5. On firebase, change numberOfPurchases to be at least 2 and nextPurchase to be <7. Make sure the color of the item is now red.
6. On firebase, change numberOfPurchases to be at least 2 and nextPurchase to be between 7 & 30. Make sure the color of the item is now orange.
7. On firebase, change numberOfPurchases to be at least 2 and nextPurchase to be more than 30. Make sure the color of the item is now green.
8. On firebase, pick an item with at least 2 numberOfPurchases and the lowest nextPurchase (e.g 2). Change the lastPurchaseDate to be more than 2 times of the item's nextPurchase (e.g. if nextPurchase is 2 then change lastPurchaseDate to 4 days ago or more). Make sure the color of the item is now grey (inactive item).
